tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -175,7 +185,66 @@ static int spi_null_tx_buf(struct device *dev, struct spi_config *spi_conf)
 
 	int ret;
 
-	LOG_INF("Start");
+	LOG_INF("Start null tx");
+
+	ret = spi_transceive(dev, spi_conf, &tx, &rx);
+	if (ret) {
+		LOG_ERR("Code %d", ret);
+		zassert_false(ret, "SPI transceive failed");
+		return ret;
+	}
+
+
+	if (memcmp(buffer_rx, EXPECTED_NOP_RETURN_BUF, BUF_SIZE)) {
+		to_display_format(buffer_rx, BUF_SIZE, buffer_print_rx);
+		LOG_ERR("Rx Buffer should contain NOP frames but got: %s",
+			buffer_print_rx);
+		zassert_false(1, "Buffer not as expected");
+		return -1;
+	}
+
+	LOG_INF("Passed null tx");
+
+	return 0;
+}
+
+#ifdef CONFIG_SPI_STM32_DMA
+static int spi_null_rx_buf(struct device *dev, struct spi_config *spi_conf)
```

Comment:
This test seems useless. You send  buffer_rx, which name is against what the test is doing, and verify that it's content is still what you have put inside?

If the test is supposed to test null_rx, it will be like testing spi_write(). No need for any rx related spi_buf/spi_buf_set (and it will not be STM32_DMA specific at all also)

Reply:
spi_null_rx_buf removed